modules/authentication/src/handlers/local.ts

```
@@ -756,6 +775,60 @@ export class LocalHandlers implements IAuthenticationStrategy {
     return 'Email changed successfully';
   }
 
+  async resendVerificationEmail(
+    call: ParsedRouterRequest,
+  ): Promise<UnparsedRouterResponse> {
+    let { email } = call.request.params;
+    if (AuthUtils.invalidEmailAddress(email)) {
+      throw new GrpcError(status.INVALID_ARGUMENT, 'Invalid email address provided');
+    }
+    email = email.toLowerCase();
+    const user: User | null = await User.getInstance().findOne({
+      email: email,
+    });
+    if (isNil(user)) throw new GrpcError(status.NOT_FOUND, 'User not found');
+    if (user.isVerified) return 'User is already verified';
```

This should throw an error instead.